### ISSUE TITLE
Crash on field request for fields='redcap_event_name'

### ISSUE THREAD

**Issue by spgarbet** [vubiostat/redcapAPI#102]
Is this expected?

```
> exportRecordsTyped(test_conn, fields=c("redcap_event_name"))
 Error in exportRecordsTyped(test_conn, fields = c("redcap_event_name")) : 
1 assertions failed:
* Variable 'fields': Must be a subset of
* {'record_id','date_dmy','date_mdy','date_ymd','datetime_dmy_hm','datetime_mdy_hm','datetime_ymd_hm','datetime_dmy_hms','datetime_mdy_hms','datetime_ymd_hms','time_hhmmss','time_hhmm','time_mmss','fieldtovar_datetimes_complete','treatment','randomization_complete','row_purpose','prereq_radio','prereq_checkbox','prereq_number','prereq_date','prereq_yesno','no_prereq_number','no_prereq_checkbox','one_prereq_non_checkbox','one_prereq_checkbox','two_prereq_and','two_prereq_or','two_prereq_and_one_check','three_prereq_andor','one_prereq_inequality','branching_logic_complete','left_operand','right_operand','calc_addition','calc_squared','text_test','date_dmy_test','date_mdy_test','date_ymd_test','datetime_dmy_hm_test','datetime_mdy_hm_test','datetime_ymd_hm_test','datetime_dmy_hms_test','datetime_mdy_hms_test','datetime_ymd_hms_test','email_test','integer_test','letters_only_test','number_test','number_1dp_comma_test',
```

**Comment by nutterb** [comment]
Um...I'm not surprised by it? But that doesn't mean it should happen.

Adding this line https://github.com/nutterb/redcapAPI/blob/102-fail-on-system-field/R/exportRecordsTyped.R#L299 avoids the error. 

In this particular case, the result took me by surprise. Whereas `fields` only contained a system field, by removing it we end up with `fields = character(0)`. Zero-length arguments don't get passed to the API, resulting in all fields being returned. 

The literalist in me thinks that the user requested only a subset of fields, they should only get a subset of fields. The pragmatist is recognizing that someone that is only requesting system fields isn't getting a useful data set and should probably rethink their request. Any thoughts on how that should be handled?

**Comment by nutterb** [comment]
Case analysis (return requested only)

* Request no fields => Return all  _I agree with this; it is consistent with what the REDCap API claims will happen_
* Request only actual fields => Return requested only **_I'm less convinced of this; the documented REDCap API behavior is that will will return redcap_event_name, redcap_repeat_instrument, and redcap_repeat_instance, when appropriate. These are all required to uniquely identify a data point in longitudinal projects. Do we want to deviate from their documented behavior like this? How will that affect users with low knowledge about internal REDCap workings?_**
* Request actual fields + system fields => Return requested only _I can get behind this. If the user is informed enough to deliberately include system fields in their request, then I think we can accommodate that._
* Request only system fields => Return requested only _No objections. It's a weird request to make, but again, if the user is informed enough to ask for it..._

**Comment by spgarbet** [comment]
Fixed with #105
